appcelerator/titanium_mobile
# feat(android): add Intl DateTimeFormat, NumberFormat, and Collator

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-27240
- https://jira.appcelerator.org/browse/TIMOB-27890
- https://jira.appcelerator.org/browse/TIMOB-27891
- https://jira.appcelerator.org/browse/TIMOB-27892

**Summary:**
_(iOS already supports the below. Adding to Android for parity.)_
- Added `Intl.Collator` support.
- Added `Intl.DateTimeFormat` support.
- Added `Intl.NumberFormat` support.
- Added locale/options support to below methods. _(Used to only support 'en-US'.)_
  * `Date.toLocaleString()`
  * `Date.toLocaleDateString()`
  * `Date.toLocaleTimeString()`
  * `Number.toLocaleString()`
  * `String.localeCompare()`
  * `String.toLocaleLowerCase()`
  * `String.toLocaleUpperCase()`

---
**Test:**
1. Build and run the below on Android.
2. Set device's region/locale to US English.
3. Launch the app and verify it shows the below.
`<Month>/<Day>/<Year>`
`1,234,567.8`
`$1,234.56`
4. Set device's region/locale to Germany (aka: Deutchland).
5. Re-launch app and verify it shows the below.
`<Day>.<Month>.<Year>`
`1.234.567,8`
`1.234,56 €`
6. Set device's region/locale to Japan (aka: 日本語 (日本))
7. Re-launch app and verify it shows the below.
`<Year>/<Month>/<Day>`
`1,234,567.8`
`¥ 1,235`

app.js
```javascript
var locale = Ti.Locale.currentLocale;
var window = Ti.UI.createWindow({ layout: "vertical" });
window.add(Ti.UI.createLabel({
	text: (new Date()).toLocaleDateString(locale),
	top: Ti.App.iOS ? "30dp" : "10dp",
}));
window.add(Ti.UI.createLabel({
	text: (1234567.8).toLocaleString(locale),
	top: "10dp",
}));
window.add(Ti.UI.createLabel({
	text: (1234.56).toLocaleString(locale, {
		style: 'currency',
		currency: Ti.Locale.getCurrencyCode(locale),
	}),
	top: "10dp",
}));
window.open();
```
